resource-managers/yarn/src/main/scala/org/apache/spark/scheduler/cluster/YarnSchedulerBackend.scala

```diff
@@ -161,6 +170,27 @@ private[spark] abstract class YarnSchedulerBackend(
     totalRegisteredExecutors.get() >= totalExpectedExecutors * minRegisteredRatio
   }
 
+  override def getMergerLocations(
+      numPartitions: Int,
+      resourceProfileId: Int): Seq[BlockManagerId] = {
+    // Currently this is naive way of calculating numMergersNeeded for a stage. In future,
+    // we can use better heuristics to calculate numMergersNeeded for a stage.
+    val tasksPerExecutor = sc.resourceProfileManager
+        .resourceProfileFromId(resourceProfileId).maxTasksPerExecutor(sc.conf)
+    val numMergersNeeded = math.min(
+      math.max(1, math.ceil(numPartitions / tasksPerExecutor).toInt), maxNumExecutors)
+    val minMergersThreshold = math.max(minMergersStaticThreshold,
+      math.floor(numMergersNeeded * minMergersThresholdRatio).toInt)
```

Comment:
This variable is a bit misleading.
It is not the minimum mergers - if `numMergersNeeded` is less than `minMergersThreshold`, we will still return mergers if we obtained `numMergersNeeded` from block manager.

I am wondering if this is what was intended in this method ?

Reply:
`minMergersThreshold` calculates mergers required both from `minMergersThresholdRatio` as well as the `minMergersStaticThreshold` - this doesn't consider `numMergersNeeded`. If `numMergersNeeded` < `minMergersThreshold` we would want to return the mergers obtained from `BlockManagerMaster`. Am I missing something here?

Reply:
> If numMergersNeeded < minMergersThreshold we would want to return the mergers obtained from BlockManagerMaster. 

So if `numMergersNeeded=3` and `minMergersThreshold=5` and mergers returned from BlockManagerMaster is 100, you want to return the 100 mergers here? If so, it doesn't look reasonable.

Reply:
> So if numMergersNeeded=3 and minMergersThreshold=5 and mergers returned from BlockManagerMaster is 100, you want to return the 100 mergers here? If so, it doesn't look reasonable.

In that case would only request `numMergersNeeded` mergers to `BlockManagerMaster` which is `3` not `100`. Let me know if this clarifies it or not.

Reply:
The name `minMergersThreshold` is a bit confusing - can you rename it ?

With `minMergersThreshold `, I would expect the minimum required to be enforced irrespective of other conditions. If mergers < threshold, no push based shuffle. This could be either due to lack of available mergers, due to a very small stage or other reasons.
I am guessing the intention is not to do this, which is why more clarity in variable name should help. Thoughts ?

Reply:
Got it. Will fix it.